MCFM-JHUGen/src/ZZ/getggHZZamps.f

```diff
@@ -63,6 +64,28 @@ subroutine getggHZZamps(p,Mloop_bquark,Mloop_tquark)
       ! Overall factor=1
       !ggHmq(:,:,:) = ggHmq(:,:,:)
 
+c--- Setting Anomalous Zff Couplings 
+      if (AllowAnomalousZffCouplings .eq. 1) then
+        if ((plabel(3) .eq. 'el') .or. (plabel(3) .eq. 'ml')
+     &.or. (plabel(3) .eq. 'tl')) then
+          l1 = leZ
+          r1 = reZ 
+        endif
+        elseif (plabel(3) .eq. 'nl') then
+          l1 = lnZ*dsqrt(3d0)
+          r1 = rnZ*dsqrt(3d0) 
+        endif
+      if (AllowAnomalousZffCouplings .eq. 1) then
+        if ((plabel(5) .eq. 'el') .or. (plabel(5) .eq. 'ml')
+     &.or. (plabel(5) .eq. 'tl')) then
+          l2 = leZ
+          r2 = reZ 
+        endif
+        elseif (plabel(5) .eq. 'nl') then
+          l2 = lnZ*dsqrt(3d0)
+          r2 = rnZ*dsqrt(3d0) 
+        endif
+
```

Comment:
Insufficient set of if-conditions. Matrix elements also handle quark final states.